app/encoding.go

```diff
@@ -1,17 +1,72 @@
 package app
 
 import (
+	"github.com/cosmos/cosmos-sdk/client"
+	amino "github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/codec/types"
 	"github.com/cosmos/cosmos-sdk/simapp/params"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/tx"
 
 	"github.com/cosmos/ethermint/codec"
+	evmtypes "github.com/cosmos/ethermint/x/evm/types"
 )
 
 // MakeEncodingConfig creates an EncodingConfig for testing
 func MakeEncodingConfig() params.EncodingConfig {
-	encodingConfig := params.MakeTestEncodingConfig()
+	cdc := amino.NewLegacyAmino()
+	interfaceRegistry := types.NewInterfaceRegistry()
+	marshaler := amino.NewProtoCodec(interfaceRegistry)
+
+	encodingConfig := params.EncodingConfig{
+		InterfaceRegistry: interfaceRegistry,
+		Marshaler:         marshaler,
+		TxConfig:          NewTxConfig(marshaler),
+		Amino:             cdc,
+	}
+
 	codec.RegisterLegacyAminoCodec(encodingConfig.Amino)
 	ModuleBasics.RegisterLegacyAminoCodec(encodingConfig.Amino)
 	codec.RegisterInterfaces(encodingConfig.InterfaceRegistry)
 	ModuleBasics.RegisterInterfaces(encodingConfig.InterfaceRegistry)
 	return encodingConfig
 }
+
+type txConfig struct {
+	cdc amino.ProtoCodecMarshaler
+	client.TxConfig
+}
+
+// NewTxConfig returns a new protobuf TxConfig using the provided ProtoCodec and sign modes. The
+// first enabled sign mode will become the default sign mode.
+func NewTxConfig(marshaler amino.ProtoCodecMarshaler) client.TxConfig {
+	return &txConfig{
+		marshaler,
+		tx.NewTxConfig(marshaler, tx.DefaultSignModes),
+	}
+}
+
+// TxEncoder overwites sdk.TxEncoder to support MsgEthereumTx
+func (g txConfig) TxEncoder() sdk.TxEncoder {
```

Comment:
I wonder why the default tx encoder doesn't work?

Reply:
See https://github.com/cosmos/cosmos-sdk/blob/v0.40.0-rc3/x/auth/tx/decoder.go#L14, the input parameter of the default encoder must be `sdk.TxRaw`. Although we can convert `MsgEthereumTx` to `sdk.Tx` when broadcasting tx, this is not what we expect.

Reply:
I think the idea is to update them to support `sdk.TxRaw`, but we can do that on a separate PR